# Titanic survivors

### We proceed to analyze different characteristics of the people who survived this catastrophe.

What characteristics did the people who survived have? We are going to start by importing pandas and numpy, and the dataset that is in csv format

In [2]:
import pandas as pd
import numpy as np

In [3]:
#crate a dataframe
df= pd.read_csv('titanic_raw.csv')

In [4]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

## Dataset Exploration

### Meanings of values:
- Survived 0= no and 1= yes
- Pclass 1= fisrt class / 2= second class / 3= therd class
- Sex female/male
- SibSp 0= unaccompanied and 1= accompanied [siblings, partner on board]
- Parch 0=unaccompanied and 1= accompanied [parents with childrens on board]
- Embarked C= Cherbourg / S= Southampton / Q= Queenstown

##### First task: we are going to determine the amount of null data, and what percentage it represents over the total data.

In [5]:
df.notnull()

PassengerId  Survived  Pclass  Name   Sex    Age  SibSp  Parch  Ticket  \
0           True      True    True  True  True   True   True   True    True   
1           True      True    True  True  True   True   True   True    True   
2           True      True    True  True  True   True   True   True    True   
3           True      True    True  True  True   True   True   True    True   
4           True      True    True  True  True   True   True   True    True   
..           ...       ...     ...   ...   ...    ...    ...    ...     ...   
886         True      True    True  True  True   True   True   True    True   
887         True      True    True  True  True   True   True   True    True   
888         True      True    True  True  True  False   True   True    True   
889         True      True    True  True  True   True   True   True    True   
890         True      True    True  True  True   True   True   True    True   

     Fare  Cabin  Embarked  
0    True  False      True  
1    True   True      True  
2    True  False      True  
3    True   True      True  
4    True  False      True  
..    ...    ...       ...  
886  True  False      True  
887  True   True      True  
888  True  False      True  
889  True   True      True  
890  True  False      True  

[891 rows x 12 columns]

In [6]:
df[df.notnull()]

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [7]:
#Calculate the amount of nulls in every column using sum
df.notnull().sum()

PassengerId    891
Survived       891
Pclass         891
Name           891
Sex            891
Age            714
SibSp          891
Parch          891
Ticket         891
Fare           891
Cabin          204
Embarked       889
dtype: int64

In [8]:
#We know the rows are 891, so we have Age less data 714, same for Cabin 204 and Embark 889.
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [9]:
#Average Nulls in the 'Age' column
df.Age.isnull().mean()

0.19865319865319866

In [10]:
#Average Nulls in the 'Cabin' column
df.Cabin.isnull().mean()

0.7710437710437711

In [11]:
#Average Nulls in the 'Embarked' column
df.Embarked.isnull().mean()

0.002244668911335578

##### Analysis by gender

In [15]:
df.Sex == 'female'

0      False
1       True
2       True
3       True
4      False
       ...  
886    False
887     True
888     True
889    False
890    False
Name: Sex, Length: 891, dtype: bool

In [16]:
df[df.Sex == 'female']

PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
9             10         1       2   
..           ...       ...     ...   
880          881         1       2   
882          883         0       3   
885          886         0       3   
887          888         1       1   
888          889         0       3   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
..                                                 ...     ...   ...    ...   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
882                       Dahlberg, Miss. Gerda Ulrika  female  22.0      0   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   

     Parch            Ticket     Fare Cabin Embarked  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
8        2            347742  11.1333   NaN        S  
9        0            237736  30.0708   NaN        C  
..     ...               ...      ...   ...      ...  
880      1            230433  26.0000   NaN        S  
882      0              7552  10.5167   NaN        S  
885      5            382652  29.1250   NaN        Q  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  

[314 rows x 12 columns]

In [17]:
df[df.Sex == 'male']

PassengerId  Survived  Pclass                            Name   Sex  \
0              1         0       3         Braund, Mr. Owen Harris  male   
4              5         0       3        Allen, Mr. William Henry  male   
5              6         0       3                Moran, Mr. James  male   
6              7         0       1         McCarthy, Mr. Timothy J  male   
7              8         0       3  Palsson, Master. Gosta Leonard  male   
..           ...       ...     ...                             ...   ...   
883          884         0       2   Banfield, Mr. Frederick James  male   
884          885         0       3          Sutehall, Mr. Henry Jr  male   
886          887         0       2           Montvila, Rev. Juozas  male   
889          890         1       1           Behr, Mr. Karl Howell  male   
890          891         0       3             Dooley, Mr. Patrick  male   

      Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0    22.0      1      0         A/5 21171   7.2500   NaN        S  
4    35.0      0      0            373450   8.0500   NaN        S  
5     NaN      0      0            330877   8.4583   NaN        Q  
6    54.0      0      0             17463  51.8625   E46        S  
7     2.0      3      1            349909  21.0750   NaN        S  
..    ...    ...    ...               ...      ...   ...      ...  
883  28.0      0      0  C.A./SOTON 34068  10.5000   NaN        S  
884  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  
886  27.0      0      0            211536  13.0000   NaN        S  
889  26.0      0      0            111369  30.0000  C148        C  
890  32.0      0      0            370376   7.7500   NaN        Q  

[577 rows x 12 columns]

In [19]:
#To confirm the gender numbers 577 males, and 314 females we can use COUNT
df[df.Sex == 'female'].count()

PassengerId    314
Survived       314
Pclass         314
Name           314
Sex            314
Age            261
SibSp          314
Parch          314
Ticket         314
Fare           314
Cabin           97
Embarked       312
dtype: int64

In [20]:
df[df.Sex == 'male'].count()

PassengerId    577
Survived       577
Pclass         577
Name           577
Sex            577
Age            453
SibSp          577
Parch          577
Ticket         577
Fare           577
Cabin          107
Embarked       577
dtype: int64

In [21]:
#We can conclude that there was a majority of males over females.

##### Analysis by Age

In [28]:
#What is the maximum and minimum age?

df.Age.min()

0.42

In [29]:
df.Age.max()

80.0

In [30]:
#Average of age per gender
df.Age[df.Sex == 'female'].mean()

27.915708812260537

In [31]:
df.Age[df.Sex == 'male'].mean()

30.72664459161148

In [33]:
#How many were of legal age?
df[df.Age >18].count()

PassengerId    575
Survived       575
Pclass         575
Name           575
Sex            575
Age            575
SibSp          575
Parch          575
Ticket         575
Fare           575
Cabin          162
Embarked       573
dtype: int64

In [34]:
#How many were minors?
df[df.Age<18].count()

PassengerId    113
Survived       113
Pclass         113
Name           113
Sex            113
Age            113
SibSp          113
Parch          113
Ticket         113
Fare           113
Cabin           19
Embarked       113
dtype: int64

In [122]:
#There were 575 adults and 113 minors.

#How many childrens under 8 years old?
df[df.Age<8].count()

In [123]:
#How many persons over 60?
df[df.Age>60].count()

PassengerId    22
Survived       22
Pclass         22
Name           22
Sex            22
Age            22
SibSp          22
Parch          22
Ticket         22
Fare           22
Cabin          11
Embarked       21
dtype: int64

##### Survivor analysis

###### We want to know, What proportion of the survivors were women and children?

In [41]:
df[df.Survived == 1].count()

PassengerId    342
Survived       342
Pclass         342
Name           342
Sex            342
Age            290
SibSp          342
Parch          342
Ticket         342
Fare           342
Cabin          136
Embarked       340
dtype: int64

In [42]:
#342 persons survived. How many didn't?
df[df.Survived == 0].count()

PassengerId    549
Survived       549
Pclass         549
Name           549
Sex            549
Age            424
SibSp          549
Parch          549
Ticket         549
Fare           549
Cabin           68
Embarked       549
dtype: int64

In [48]:
df.Survived.sum() #we can calculat with SUM how many survived?

342

In [49]:
df.Survived[df.Survived == 1].count()

342

In [52]:
#What is the porcentage of the total?
round (df.Survived[df.Survived == 1].count() / df.Survived.count(),3) * 100

38.4

In [53]:
#Almost the 40% survived

In [54]:
#Were they men or women?
df[(df.Sex =='female')&(df.Survived==1)].count(0)

PassengerId    233
Survived       233
Pclass         233
Name           233
Sex            233
Age            197
SibSp          233
Parch          233
Ticket         233
Fare           233
Cabin           91
Embarked       231
dtype: int64

In [55]:
df[(df.Sex =='male')&(df.Survived==1)].count(0)

PassengerId    109
Survived       109
Pclass         109
Name           109
Sex            109
Age             93
SibSp          109
Parch          109
Ticket         109
Fare           109
Cabin           45
Embarked       109
dtype: int64

In [56]:
#233 women survived, and 109 men

In [59]:
#What was the supervivence number for the childrens?
df[(df.Age <18)&(df.Survived==1)].count()

PassengerId    61
Survived       61
Pclass         61
Name           61
Sex            61
Age            61
SibSp          61
Parch          61
Ticket         61
Fare           61
Cabin          17
Embarked       61
dtype: int64

In [60]:
#How many minors didn't survive?
df[(df.Age <18)&(df.Survived==0)].count()

PassengerId    52
Survived       52
Pclass         52
Name           52
Sex            52
Age            52
SibSp          52
Parch          52
Ticket         52
Fare           52
Cabin           2
Embarked       52
dtype: int64

In [61]:
#What was the supervivence number for the childrens under 6 years old?
df[(df.Age<6)&(df.Survived==1)].count()

PassengerId    31
Survived       31
Pclass         31
Name           31
Sex            31
Age            31
SibSp          31
Parch          31
Ticket         31
Fare           31
Cabin           7
Embarked       31
dtype: int64

In [62]:
df[(df.Age<6)&(df.Survived==0)].count() #How many didn't survive under 6yo?

PassengerId    13
Survived       13
Pclass         13
Name           13
Sex            13
Age            13
SibSp          13
Parch          13
Ticket         13
Fare           13
Cabin           2
Embarked       13
dtype: int64

## Titanic survival analysis in relative terms

In [66]:
#Let's create some variables to be able to create some metrics

In [67]:
total_women # who embraked in the Titanic

314

In [68]:
total_surv_women = df.Survived[(df.Sex =='female')&(df.Survived ==1)].count()

In [69]:
total_surv_women #women who survived

233

In [70]:
#female survival rate
total_surv_women / total_women *100

74.20382165605095

In [71]:
female_surv_rate = round(total_surv_women / total_women,3)*100

In [87]:
female_surv_rate #The female survival rate was the 74.2%

74.2

In [88]:
#How many men embaked?
df.Survived[df.Sex =='male'].count()

577

In [89]:
total_men = df.Survived[df.Sex =='male'].count()

In [90]:
total_surv_men= df.Survived[(df.Sex =='male')&(df.Survived ==1)].count()

In [91]:
total_surv_men #How many men survived?

109

In [92]:
male_surv_rate = round (total_surv_men/total_men,3)*100

In [93]:
male_surv_rate #The male survival rate was the 18.9%

18.9

In [95]:
#How many childrens under 12 embarked?
df.Age[df.Age<12].count()

68

In [133]:
total_children=df.Age[df.Age<6].count()

In [134]:
#How many survived?
df.Age[(df.Age<6)&(df.Survived==1)].count()

31

In [135]:
total_surv_children=df.Age[(df.Age<6)&(df.Survived==1)].count()

In [136]:
children_surv_rate= round (total_surv_children/total_children , 3)*100

In [137]:
children_surv_rate #The children survival rate under 12 years old was 57.4%

70.5

In [104]:
#Comparisition of supervival rates in differents boarding classes

In [105]:
df.Survived[df.Pclass ].count()

891

In [106]:
total_people = df.Survived[df.Pclass ].count()

In [107]:
df.Survived[df.Pclass ==1].count()

216

In [108]:
first_class = df.Survived[df.Pclass ==1].count()

In [109]:
df.Survived[df.Pclass ==2].count()

184

In [110]:
second_class = df.Survived[df.Pclass ==2].count()

In [141]:
third_class = df.Survived[df.Pclass ==3].count()

In [142]:
#How many people survived in first class?
df.Survived[(df.Pclass ==1)&(df.Survived==1)].count()

136

In [143]:
first_class_surv = df.Survived[(df.Pclass ==1)&(df.Survived==1)].count()

In [144]:
first_class_surv / first_class *100

62.96296296296296

In [145]:
first_class_srv_rate = round (first_class_surv / first_class,3) *100
first_class_srv_rate #What is the percentage of survival in first class?

63.0

In [146]:
second_class_surv = df.Survived[(df.Pclass ==2)&(df.Survived==1)].count()

In [147]:
second_class_surv

87

In [148]:
second_class_srv_rate = round (second_class_surv / second_class,3) *100
second_class_srv_rate #What is the percentage of survival in second class?

47.3

In [149]:
third_class_surv = df.Survived[(df.Pclass ==3)&(df.Survived==1)].count()

In [150]:
third_class_surv 

119

In [151]:
third_class_srv_rate = round (third_class_surv / third_class,3) *100
third_class_srv_rate #What is the percentage of survival in third class?

24.2

In [152]:
total_surv_people = df.Survived[df.Survived ==1].count()
total_surv_people

342

In [127]:
rate_survival = round(total_surv_people / total_people,3) *100
rate_survival #General survival rate

38.4

In [129]:
total_adults=df.Age[df.Age<18].count()
total_adults #How many adults over 18?

113

In [131]:
#How many people over 18 survived?
total_surv_adults=df.Age[(df.Age<18)&(df.Survived==1)].count()
total_surv_adults

61

In [132]:
Adults_surv_rate = round (total_surv_adults/total_adults , 3) *100
Adults_surv_rate

54.0

# Analysis conclusions

### Some insights:

##### Of 891 persons on the Titanic:

There were 575 adults and 113 minors under 18 years old.

There were 314 women and 577 men.

342 persons survived and 549 didn't.


The dataset had some Nulls: 19.8% of Age data, and  77.10% of Cabin data


The youngest was 0,4 years old (less than 1 y ar) and the oldest was 80 years o
d.

The average age of females was 27 years old and males 30 years old.ears old.

### Some metrics:

#### The general survival rate was 38.4%

##### Gender metrics:

Female's survival rate was 74.2%.

Male's survival rate was 18.9%.

"Women had a better chance of surviving."


##### Age Metrics:

Children's survival rate was 57.4% (under 12 years old).

Children's survival rate was 70.5% (under 6 years old).

Adults's survival rate was 54% (over 18 years old).

"The younger you were, the more likely you were to be a survivor."

##### Boarding class metrics:

First-class survival rate was 63%.

Second-class survival rate was 47.3
%
Third-class survival rate was 24.2

"The higher the boarding pass class, the more likely you were to survive."%.